sched/sched/sched_timerexpiration.c

```diff
@@ -494,19 +496,19 @@ void nxsched_timer_expiration(void)
 
   /* Get the interval associated with last expiration */
 
-  flags = enter_critical_section();
+  flags = spin_lock_irqsave_wo_note(&g_lock);
   up_timer_gettick(&ticks);
   g_timer_tick = ticks;
   elapsed = g_timer_interval;
-  leave_critical_section(flags);
+  spin_unlock_irqrestore_wo_note(&g_lock, flags);
 
   /* Process the timer ticks and set up the next interval (or not) */
 
   nexttime = nxsched_timer_process(ticks, elapsed, false);
 
-  flags = enter_critical_section();
+  flags = spin_lock_irqsave_wo_note(&g_lock);
   g_timer_interval = nxsched_timer_start(ticks, nexttime);
```

Comment:
`nxsched_timer_start()` will access the global variables in lower harlf driver, using local spinlock cannot guarantee the same effect as csection.

Reply:
nxsched_timer_start will call up_timer_start or up_alarm_start 
I believe that these two functions should be responsible for adding their own critical sections. In fact, this is probably how it's done in most cases.
![image](https://github.com/user-attachments/assets/41015082-8fff-4c4b-87be-47228b62d948)
![image](https://github.com/user-attachments/assets/66de2bd2-24ae-4b93-a74f-46d0fa62d0e1)

However, there may be some overlooked scenarios. Currently, we can add an additional critical section in this function. Once all the timer driver codes have been replaced with small locks in the future, this critical section can be removed.

Reply:
![image](https://github.com/user-attachments/assets/77cf5a25-943a-4aee-a9b2-125f51cf4bce)